docs/rust-on-xtensa-installation-aarch64-apple-darwin.md

```diff
@@ -1,35 +1,69 @@
 # Rust on Xtensa Installation for macOS M1
 
+Following instructions are specific for ESP32 and ESP32-S series based on Xtensa architecture.
+
+Instructions for ESP-C series based on RISC-V architecture are described in document for [ESP32-C3](../README.md#esp32-c3).
+
 Tested OS: macOS Big Sur M1
 
 ## Prerequisites
 
 - rustup - https://rustup.rs/
 
-## Commands
+## Installation commands
 
-```
+```sh
 rustup toolchain install nightly
 
-mkdir -p ~/.rustup/toolchains/esp
+VERSION="1.54.0-dev"
+ARCH="aarch64-apple-darwin"
+RUST_DIST="rust-${VERSION}-${ARCH}"
+RUST_SRC_DIST="rust-src-${VERSION}"
+TOOLCHAIN_DESTINATION_DIR="~/.rustup/toolchains/esp"
+
+mkdir -p ${TOOLCHAIN_DESTINATION_DIR}
 
-wget --continue https://dl.espressif.com/dl/idf-rust/dist/aarch64-apple-darwin/rust-1.53.0-dev-aarch64-apple-darwin.tar.xz
-tar xvf rust-1.53.0-dev-aarch64-apple-darwin.tar.xz
-./rust-1.53.0-dev-aarch64-apple-darwin/install.sh --destdir=~/.rustup/toolchains/esp --prefix="" --without=rust-docs
+wget "https://dl.espressif.com/dl/idf-rust/dist/${ARCH}/${RUST_DIST}.tar.xz"
+tar xvf ${RUST_DIST}.tar.xz
+./${RUST_DIST}/install.sh --destdir=${TOOLCHAIN_DESTINATION_DIR} --prefix="" --without=rust-docs
 
-wget --continue https://dl.espressif.com/dl/idf-rust/dist/noarch/rust-src-1.53.0-dev.tar.xz
-tar xvf rust-src-1.53.0-dev.tar.xz
-./rust-src-1.53.0-dev/install.sh --destdir=~/.rustup/toolchains/esp --prefix=""
+wget "https://dl.espressif.com/dl/idf-rust/dist/noarch/${RUST_SRC_DIST}.tar.xz"
+tar xvf ${RUST_SRC_DIST}.tar.xz
+./${RUST_SRC_DIST}/install.sh --destdir=${TOOLCHAIN_DESTINATION_DIR} --prefix="" --without=rust-docs
 
 rustup default esp
```

Comment:
Perhaps we should remove this step; because we don't ship all components first time rust users can run into issues where they can't install other components they may need [source](https://matrix.to/#/!LdaNPfUfvefOLewEIM:matrix.org/$eagAo3CVVfSRQp2tscqNRI2RdOHg9ug6jzbcvAJiOZQ?via=matrix.org&via=tchncs.de&via=finallycoffee.eu). 

Instead we should add a `rust-toolchain` file in the root with the value of `esp`

Reply:
I made small investigation of this option. It's now applied in Windows distribution, because there is no other way and there is no installer.  I will test this approach for macOS and Linux in next round of building compilers.